src/OpenApi/src/Extensions/OpenApiEndpointRouteBuilderExtensions.cs

```diff
@@ -0,0 +1,57 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Diagnostics.CodeAnalysis;
+using System.Globalization;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.OpenApi;
+using Microsoft.AspNetCore.Routing;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Options;
+using Microsoft.OpenApi.Extensions;
+using Microsoft.OpenApi.Writers;
+
+namespace Microsoft.AspNetCore.Builder;
+
+/// <summary>
+/// OpenAPI-related methods for <see cref="IEndpointRouteBuilder"/>.
+/// </summary>
+public static class OpenApiEndpointRouteBuilderExtensions
+{
+    /// <summary>
+    /// Register an endpoint onto the current application for resolving the OpenAPI document associated
+    /// with the current application.
+    /// </summary>
+    /// <param name="endpoints">The <see cref="IEndpointRouteBuilder"/>.</param>
+    /// <param name="pattern">The route to register the endpoint on. Must include the 'documentName' route parameter.</param>
+    /// <returns>An <see cref="IEndpointRouteBuilder"/> that can be used to further customize the endpoint.</returns>
+    public static IEndpointConventionBuilder MapOpenApi(this IEndpointRouteBuilder endpoints, [StringSyntax("Route")] string pattern = OpenApiConstants.DefaultOpenApiRoute)
```

Comment:
Nit:

```suggestion
    public static IEndpointConventionBuilder MapOpenApi(this IEndpointRouteBuilder endpoints, [StringSyntax("Route")] string pattern = "/openapi/{documentName}.json")
```

I prefer being able to see the default route inline without having to go to yet another file.

Reply:
I've gotten mixed feedback on the use of constants here. Given that you can view the value in the editor hint, I think this is fine if we use the constant here. It is a bit of a degrade experience if you're reading the code in something like GitHub but I think go-to def works well for those cases.